nipype/interfaces/afni/preprocess.py

```diff
@@ -3722,3 +3621,80 @@ def _list_outputs(self):
     def _gen_filename(self, name):
         if name == 'out_file':
             return self._gen_fname(self.inputs.source_file, suffix='_QW')
+
+
+class QwarpPlusMinusInputSpec(QwarpInputSpec):
+    in_file = File(
+        desc='Source image (opposite phase encoding direction than base image)',
+        argstr='-source %s',
+        mandatory=True,
+        exists=True,
+        copyfile=False)
+    source_file = File(
+        desc='Source image (opposite phase encoding direction than base image)',
+        argstr='-source %s',
+        exists=True,
+        deprecated='1.1.2',
+        new_name='in_file',
+        copyfile=False)
+    out_file = File(
+        argstr='-prefix %s',
+        value='Qwarp.nii.gz',
+        position=0,
+        usedefault=True,
+        desc="Output file")
+    plusminus = traits.Bool(
+        True,
+        usedefault=True,
+        position=1,
+        desc='Normally, the warp displacements dis(x) are defined to match'
+        'base(x) to source(x+dis(x)).  With this option, the match'
+        'is between base(x-dis(x)) and source(x+dis(x)) -- the two'
+        'images \'meet in the middle\'. For more info, view Qwarp` interface',
+        argstr='-plusminus',
+        xor=['duplo', 'allsave', 'iwarp'])
+
+
+class QwarpPlusMinusOutputSpec(QwarpOutputSpec):
+    warped_source = File(desc='Undistorted source file.', exists=True)
+    warped_base = File(desc='Undistorted base file.', exists=True)
+    source_warp = File(
+        desc="Field suceptibility correction warp (in 'mm') for source image.",
+        exists=True)
+    base_warp = File(
+        desc="Field suceptibility correction warp (in 'mm') for base image.",
+        exists=True)
+
+
+class QwarpPlusMinus(Qwarp):
+    """A version of 3dQwarp for performing field susceptibility correction
+    using two images with opposing phase encoding directions.
+
+    For complete details, see the `3dQwarp Documentation.
+    <https://afni.nimh.nih.gov/pub/dist/doc/program_help/3dQwarp.html>`_
+
+    Examples
+    ========
+
+    >>> from nipype.interfaces import afni
+    >>> qwarp = afni.QwarpPlusMinus()
+    >>> qwarp.inputs.source_file = 'sub-01_dir-LR_epi.nii.gz'
+    >>> qwarp.inputs.nopadWARP = True
+    >>> qwarp.inputs.base_file = 'sub-01_dir-RL_epi.nii.gz'
+    >>> qwarp.cmdline
+    '3dQwarp -prefix Qwarp.nii.gz -plusminus -base sub-01_dir-RL_epi.nii.gz -nopadWARP -source sub-01_dir-LR_epi.nii.gz'
+    >>> res = warp.run()  # doctest: +SKIP
+
+    """
+
+    input_spec = QwarpPlusMinusInputSpec
+    output_spec = QwarpPlusMinusOutputSpec
+
+    def _list_outputs(self):
+        outputs = self.output_spec().get()
+        outputs['warped_source'] = os.path.abspath("Qwarp_PLUS.nii.gz")
+        outputs['warped_base'] = os.path.abspath("Qwarp_MINUS.nii.gz")
+        outputs['source_warp'] = os.path.abspath("Qwarp_PLUS_WARP.nii.gz")
+        outputs['base_warp'] = os.path.abspath("Qwarp_MINUS_WARP.nii.gz")
```

Comment:
I would drop `_list_outputs`. This should be handled by `Qwarp._list_outputs()` and correctly handle non-default `out_file` selections.